File: app/src/main/java/nolambda/stiker/motionviews/serialized/ImageEntitySavedState.java
package nolambda.stiker.motionviews.serialized;

import nolambda.stiker.motionviews.viewmodel.Layer;
import nolambda.stiker.motionviews.widget.entity.ImageEntity;

public class ImageEntitySavedState extends BaseEntitySavedState {
    private final ImageEntity.BitmapProvider bitmapProvider;

    public ImageEntitySavedState(Layer layer,
                                 float[] matrixValues,
                                 float holyScale,
                                 int canvasWidth,
                                 int canvasHeight,
                                 PaintData borderPaintData,
                                 ImageEntity.BitmapProvider bitmapProvider) {
        super(layer, matrixValues, holyScale, canvasWidth, canvasHeight, borderPaintData);
        this.bitmapProvider = bitmapProvider;
    }

    public ImageEntity.BitmapProvider getBitmapProvider() {
        return bitmapProvider;
    }
}

File: app/src/main/java/nolambda/stiker/motionviews/serialized/ImageEntitySavedState.kt
package nolambda.stiker.motionviews.serialized

import nolambda.stiker.motionviews.viewmodel.Layer
import nolambda.stiker.motionviews.widget.entity.ImageEntity

class ImageEntitySavedState(
    layer: Layer,
    matrixValues: FloatArray,
    holyScale: Float,
    canvasWidth: Int,
    canvasHeight: Int,
    borderPaintData: PaintData,
    val bitmapProvider: ImageEntity.BitmapProvider
) : BaseEntitySavedState(layer, matrixValues, holyScale, canvasWidth, canvasHeight, borderPaintData)
